lib/AST/ASTPrinter.cpp

```diff
@@ -4785,12 +4787,27 @@ void PrintAST::visitErrorExpr(ErrorExpr *expr) {
   Printer << "<error>";
 }
 
-void PrintAST::visitTernaryExpr(TernaryExpr *expr) {}
+void PrintAST::visitTernaryExpr(TernaryExpr *expr) {
+  if (auto condExpr = expr->getCondExpr()) {
+    visit(expr->getCondExpr());
+  }
+  Printer << " ? ";
+  visit(expr->getThenExpr());
+  Printer << " : ";
+  if (auto elseExpr = expr->getElseExpr()) {
+    visit(expr->getElseExpr());
+  }
+}
 
 void PrintAST::visitIsExpr(IsExpr *expr) {
+  visit(expr->getSubExpr());
+  Printer << " is ";
+  printType(expr->getCastType());
 }
 
 void PrintAST::visitTapExpr(TapExpr *expr) {
+  printType(expr->getVar()->getType());
+  printBraceStmt(expr->getBody());
```

Comment:
`TapExpr` is a tricky beast because it contains a bunch of synthesized code that we probably don't want to print (I think ideally we'd just print an interpolated string literal as it appears in source, not as a chain of `append...` calls). There may be value in showing the `append...` calls, but I'm not sure it should be the default behavior. Additionally, without `do` expressions, we'd also have to resort to printing it as an immediately-executed closure.

To properly handle this, I think we'd need to jump through all the synthesized code to find the argument nodes for the `append...` calls, and use those to reconstruct the string literal segments and interpolated expressions. I'm okay if you want to leave this unimplemented for now.